framework/mongo/utils.py

```diff
@@ -63,10 +63,18 @@ def wrapper(cls):
 
 def get_or_http_error(Model, pk):
     instance = Model.load(pk)
+
     if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
+        if hasattr(instance, 'registrations'):
```

Comment:
Repeating this exception makes it more likely that in the future, one copy might get changed and the other overlooked. I suggest something like this:

```python
if hasattr(instance, 'registrations'):
    exc_options = {
        'message_short': 'Project Deleted',
        'message_long': 'This project has been deleted',
    }
else:
    exc_options = {
        'message_long'='This resource has been deleted'
    }

raise HTTPError(http.GONE, data=exc_options)
```

Before making changes, though, I'd like to clarify whether or not this is a desirable change.